packages/chevrotain/src/parse/cst/cst.ts

```diff
@@ -23,7 +23,7 @@ export function setNodeLocationOnlyOffset(
   // any farther updates as the Token vector is sorted.
   // We still have to check this this condition for every new possible location info
   // because with error recovery enabled we may encounter invalid tokens (NaN location props)
-  else if (currNodeLocation.endOffset < newLocationInfo.endOffset === true) {
+  else if (currNodeLocation.endOffset! < newLocationInfo.endOffset! === true) {
```

Comment:
this could be confusing
`newLocationInfo.endOffset! === true` vs `newLocationInfo.endOffset !== true` 😢

Reply:
I guess? The `=== true` is redundant here anyways.

Reply:
We could do a < b === true === true to be extra sure 😆

Reply:
It is likely a performance optimization as this is a hot spot.
So we are avoiding the automatic boolean casting

Reply:
I don't think there is any auto conversion there. `<` always returns boolean afaik. (as does the isNaN above)

Reply:
It's a parser toolkit, I'm expecting contributors to know `! ===` and  `!==` parse as different tokens 😁 .

We can remove it or change it to `true === `, up to you 👍

Reply:
It is not a huge thing, I was just commenting on different things in TypeScript with syntax similarities but different meanings.